libbs/decompilers/ghidra/hooks.py

```diff
@@ -50,22 +51,49 @@ def domainObjectChanged(self, ev):
                 obj = record.getObject()
 
                 if changeType in self.funcEvents:
-                    pass
+                    subType = record.getSubEventType()
+                    if subType == 5:
+                        # Function return type changed
+                        header = FunctionHeader(None, None, str(obj.getReturnType()))
+                        self._interface.function_header_changed(header)
+
                 elif changeType in self.typeEvents:
-                    try:
-                        struct = self._interface.structs[newValue.name]
-                        # TODO: access old name indicate deletion
-                        #self._interface.struct_changed(Struct(None, None, None), deleted=True)
-                        self._interface.struct_changed(struct)
-                    except KeyError:
-                        pass
+                    if changeType == self.changeManager.DOCR_SYMBOL_ADDRESS_CHANGED:
+                        # stack variables change address when retyped!
+                        if self._interface.ghidra.isinstance(obj, self.db.function.VariableDB):
+                            parent_namespace = obj.getParentNamespace()
+                            storage = obj.getVariableStorage()
+                            if (
+                                (newValue is not None) and (storage is not None) and bool(storage.isStackStorage())
+                                and (parent_namespace is not None)
+                            ):
+                                sv = StackVariable(
+                                    int(storage.stackOffset),
+                                    None,
+                                    str(obj.getDataType()),
+                                    int(storage.size),
+                                    int(obj.parentNamespace.entryPoint.offset)
+                                )
+                                self._interface.stack_variable_changed(
+                                    sv
+                                )
+
+                    else:
+                        try:
+                            struct = self._interface.structs[newValue.name]
+                            # TODO: access old name indicate deletion
+                            #self._interface.struct_changed(Struct(None, None, None), deleted=True)
```

Comment:
Why again is this commented out? Did we have an Issue for this?

Reply:
Yeah this was the thing where we left an issue on the ghidra github page about iirc. The main problem is that once the struct changes, there is no way to know what the old struct was